### DISCUSSION TITLE
Data is automatically updating

### DISCUSSION THREAD

**Discussion by ikbal-nayem** [komarovalexander/ka-table#149] [General]
Whenever I make a copy of `tableProps.data` and change some value of that copied data then `tableProps.data` values are also change. And vice-versa
I tried this also,
`const data = [...tableProps.data]`
But no luck :(
How can I make an individual copy of `tableProps.data`

**Comment by komarovalexander** [comment]
Hi @ikbal-nayem you have to make deep copy of the array
For example if your data consists of plain object:
 ```
const data = tableProps.data.map(d=> ({...d}))
```
Or like this https://javascript.plainenglish.io/how-to-deep-copy-objects-and-arrays-in-javascript-7c911359b089
